scripts/cluster-upgrade.sh

```diff
@@ -118,33 +120,37 @@ setup() {
     MS_REPLICAS=$(($DESIRED_REPLICAS/$ZONE_COUNT))
     for MS_NAME in $(KUBECONFIG=$TMP_DIR/kubeconfig-${CD_NAMESPACE} oc -n openshift-machine-api get machineset --no-headers | grep worker | awk '{print $1}');
     do
-        log $OCM_NAME "setup" "waiting for replicas, machineset=$MS_NAME"
         AVAILABLE_REPLICAS=$(KUBECONFIG=$TMP_DIR/kubeconfig-${CD_NAMESPACE} oc -n openshift-machine-api get machineset $MS_NAME -o jsonpath='{.status.availableReplicas}')
 
         while [ "$AVAILABLE_REPLICAS" != "$MS_REPLICAS" ];
         do
-            sleep 15
+            log $OCM_NAME "setup" "waiting for replicas, machineset=$MS_NAME"
+
+            # sign CSRs, sometimes new node provisioning is stuck because of this
+            # https://github.com/openshift/ops-sop/blob/master/v4/alerts/NoNodeObjectForMachineCriticalSRE.md
+            for CSR in $(KUBECONFIG=$TMP_DIR/kubeconfig-${CD_NAMESPACE} oc get csr --no-headers 2>/dev/null | grep Pending | awk '{print $1}');
+            do
+                KUBECONFIG=$TMP_DIR/kubeconfig-${CD_NAMESPACE} oc adm certificate approve $CSR
+            done
+
+            sleep 30
 
             AVAILABLE_REPLICAS=$(KUBECONFIG=$TMP_DIR/kubeconfig-${CD_NAMESPACE} oc -n openshift-machine-api get machineset $MS_NAME -o jsonpath='{.status.availableReplicas}')
         done
 
         log $OCM_NAME "setup" "replicas are ready for upgrade, machineset=$MS_NAME"
     done
 
-    # Maint window duration is calculated based on number of nodes.
-    # A standard cluster upgrade (PD_MAINT_BASE_NODE_COUNT workers) takes PD_MAINT_BASE_MIN minutes.
-    # For every additional worker (including for upgrade capacity)
-    # we add PD_MAINT_ADDITIONAL_NODE_MIN minutes to the base window.
-    MAINT_WINDOW_MIN=$((PD_MAINT_BASE_MIN+(DESIRED_REPLICAS-PD_MAINT_BASE_NODE_COUNT)*PD_MAINT_ADDITIONAL_NODE_MIN))
-
-    setup_maintenance_window $OCM_NAME $CD_NAMESPACE $CD_NAME $MAINT_WINDOW_MIN
+    # Set the initial maintenance window to the base so control plane issues will page earlier for very large clusters.
+    setup_maintenance_window $OCM_NAME $CD_NAMESPACE $CD_NAME $PD_MAINT_BASE_MIN "$PD_MAINT_DESCRIPTION_BASE"
 }
 
 setup_maintenance_window() {
     OCM_NAME=$1
     CD_NAMESPACE=$2
     CD_NAME=$3
     MAINT_WINDOW_MIN=$4
+    MAINT_DESC=$5
```

Comment:
Do you need to quote this assignment so that multiple words don't overlap into `$6`, `$7`, etc?

Reply:
When passed to this function, yes.  Hit that in early iterations, all args to calling setup and teardown maint window functions are quoted.